src/collections/src/NICollectionViewModel.h

```diff
@@ -29,20 +29,19 @@
 @class NICollectionViewModelFooter;  // Provides the information for a footer.
 
 /**
- * A non-mutable collection view model that complies to the UICollectionViewDataSource protocol.
+ * A protocol that declares the interface for a non-mutable collection view model.
  *
- * This model allows you to easily create a data source for a UICollectionView without having to
- * implement the UICollectionViewDataSource methods in your controller.
+ * A default implementation of this protocol is provided with the NICollectionViewModel class.
+ * If you want to customize the implementation of your collection view model while keeping the base
+ * interface the same, conform to this protocol and implement the declared methods at minimum.
  *
- * This base class is non-mutable, much like an NSArray. You must initialize this model with
- * the contents when you create it.
- *
- * This model simply manages the data relationship with your collection view. It is up to you to
- * implement the collection view's layout object.
+ * The model class that conforms to this protocol is intended to allow you to easily create a data
+ * source for a UICollectionView without having to implement the UICollectionViewDataSource methods
+ * in your controller.
  *
  * @ingroup CollectionViewModels
  */
-@interface NICollectionViewModel : NSObject <NIActionsDataSource, UICollectionViewDataSource>
+@protocol NICollectionViewModel <NIActionsDataSource, UICollectionViewDataSource>
```

Comment:
I don't recall the ramifications of a type existing with the same name as both a protocol and an interface for Swift code, but my rough expectation is that this will cause problems.

Barring confirmation of that fact, I may recommend that we name the protocol `NICollectionViewModeling`, following the recommended gerund naming practice: https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/NamingBasics.html#//apple_ref/doc/uid/20001281-1002242-BBCIJGDB

Reply:
Ahh you're right! Renamed to follow the convention mentioned in the linked Apple doc.